### PR TITLE
EUI-6408: Added copy functionality

### PR DESCRIPTION
**Pull request by Surekha-HMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-6408

### Change description ###

Added copy functionality

### CODE REVIEW

File: src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts [hunk 1]
```diff
@@ -269,35 +273,92 @@ export class StaffAddEditUserFormComponent implements OnInit {
...
+          lastName: null,
+          email: null,
+          region: selectedRegion,
+          'user-services': this.getSelectedByKey(this.staffFilterOptions.services, copyUser.services),
+          jobTitle: this.getSelectedByKey(this.staffFilterOptions.jobTitles, copyUser.role.map(role => role.role_id)),
+          roles: this.getSelectedRoles(copyUser.case_allocator, copyUser.task_supervisor, copyUser.staff_admin),
+          'user-skills': this.getSelectedSkills(this.staffFilterOptions.skills, copyUser.skills),
+          userType: this.getSelectedByName(this.staffFilterOptions.userTypes, copyUser.userType),
         });
+        console.log(copyUser);
```
Comment (connorpgpmcelroy) [inline src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts:297]: Log here to remove

File: src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts [hunk 2]
```diff
@@ -269,35 +273,92 @@ export class StaffAddEditUserFormComponent implements OnInit {
...
+
+        const additionalLocations = copyUser.additionalLocations;
+        const primaryLocation = copyUser.primaryLocation
```
Comment (connorpgpmcelroy) [inline src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts:300]: Very minor comment - semi-colon

File: src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts [hunk 3]
```diff
@@ -269,35 +273,92 @@ export class StaffAddEditUserFormComponent implements OnInit {
...
+        if(additionalLocations.length > 0) {
+          const addLoc = (frm.controls['additionalLocations'] as FormArray);
+          addLoc.push(new FormControl(this.getSelectedLocation(this.staffFilterOptions.locations, copyUser.additionalLocations)));
+        }
+        if(primaryLocation) {
+          const primLoc = (frm.controls['primaryLocation'] as FormArray);
+          primLoc.push(new FormControl(this.getSelectedLocation(this.staffFilterOptions.locations, [copyUser.primaryLocation])));
+        }
+        console.log(frm);
```
Comment (connorpgpmcelroy) [inline src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts:309]: Remove log

File: src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts [hunk 4]
```diff
@@ -269,35 +273,92 @@ export class StaffAddEditUserFormComponent implements OnInit {
...
+    let staff_admin_flag = false;
+    if(selectedCaseAllocator === 'Y') {
+      case_allocator_flag = true;
+    }
+    if(selectedTaskSupervisor === 'Y') {
+      task_supervisor_flag = true;
+    }
+    if(selectedStaffAdmin === 'Y') {
+      staff_admin_flag = true;
+    }
```
Comment (connorpgpmcelroy) [inline src/staff-administrator/components/staff-add-edit-user/staff-add-edit-user-form/staff-add-edit-user-form.component.ts:351]: Above could be done in ternary operators with less lines - minor comment

File: src/staff-administrator/components/staff-user-details/staff-user-details.component.ts [hunk 5]
```diff
@@ -9,18 +9,36 @@ import { StaffDataAccessService } from '../../services/staff-data-access/staff-d
   templateUrl: './staff-user-details.component.html',
   styleUrls: ['./staff-user-details.component.scss']
 })
-export class StaffUserDetailsComponent {
+export class StaffUserDetailsComponent implements OnInit {
   public userDetails: StaffUser;
   public showAction: boolean = false;
   public loading = false;
   public suspendedStatus: 'suspended' | 'restored' | 'error';
+  selectedRoles: string[] = [];
```
Comment (connorpgpmcelroy) [inline src/staff-administrator/components/staff-user-details/staff-user-details.component.ts:17]: public selectedRoles right?

File: src/staff-administrator/models/staff-user.model.ts [hunk 6]
```diff
@@ -14,17 +14,29 @@ export interface StaffUser {
   primaryLocation?: StaffLocation;
   primaryRole?: any;
   roles: string[];
-  skills?: string[];
+  role?: StaffRole[] ;
```
Comment (connorpgpmcelroy) [inline src/staff-administrator/models/staff-user.model.ts:17]: Very minor comment - semi-colon out of position

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Lots of little changes requested

File: src/staff-administrator/resolvers/staff-filter-options-locations.resolver.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  providedIn: 'root'
+})
+export class StaffFilterOptionsLocationsResolver implements Resolve<LocationByEPIMMSModel[]> {
+
+  constructor(private readonly http: HttpClient) {
+  }
+
+  public resolve(route?: ActivatedRouteSnapshot): Observable<LocationByEPIMMSModel[]> {
+
+    return this.http.post<LocationByEPIMMSModel[]>(`api/locations/getLocations`, {serviceIds: ['SSCS','IA'], locationType: 'case-management', searchTerm: 'cen'});
```
Comment (connorpgpmcelroy) [inline src/staff-administrator/resolvers/staff-filter-options-locations.resolver.ts:18]: Extra change/query - we're setting services to only be SSCS/IA - can we edit this to take the supportedjurisdictions or get details from the user?

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Query on serviceIds being set

**Review by DavidJayakumar — APPROVED** [review]
Lgtm

**Pull request merged**